homeassistant/components/media_player/songpal.py

```diff
@@ -169,18 +243,28 @@ def available(self):
 
             inputs = await self.dev.get_inputs()
             _LOGGER.debug("Got ins: %s", inputs)
-            self._sources = inputs
+
+            self._sources = OrderedDict()
+            for input_ in inputs:
+                self._sources[input_.uri] = input_
+                if input_.active:
+                    self._active_source = input_
+
+            _LOGGER.debug("Active source: %s", self._active_source)
 
             self._available = True
+
+            # activate notifications if wanted
+            if not self._poll:
```

Comment:
Can we remove this check?

Reply:
See above. If you think it's better to remove it, I'll do that.